### PR TITLE
dont use empty string as null value

### PR DESCRIPTION
**Pull request by NilsBacke**

### CODE REVIEW

**Comment by NilsBacke** [comment]
[Make "missing" string values be undefined, not empty string](https://trello.com/c/kUYMzW3B/231-make-missing-string-values-be-undefined-not-empty-string)

File: frontend/src/Onboarding/RedirectScreen.tsx [hunk 1]
```diff
@@ -55,14 +55,11 @@ interface Props {
 
 export const RedirectScreen: React.FC<Props> = ({ redirectUrl }) => {
   const dispatch = useDispatch();
-  const { academicYear, graduationYear, isAdvisor } = useSelector(
-    (state: AppState) => ({
-      academicYear: getAcademicYearFromState(state),
-      graduationYear: getGraduationYearFromState(state),
-      isAdvisor: getIsAdvisorFromState(state),
-    })
-  );
```
Comment (NilsBacke) [inline frontend/src/Onboarding/RedirectScreen.tsx:64]: Getting these values here would error, as there is no user in our redux store at this point in the flow

File: frontend/src/state/reducers/userReducer.ts [hunk 2]
```diff
@@ -18,34 +18,14 @@ import {
...
-  sumCreditsFromList,
-  numToTerm,
-  getNextTerm,
-  produceWarnings,
-} from "../../utils";
 
 export interface UserState {
-  user: IUserData;
+  user?: IUserData;
   completedRequirements: IRequiredCourse[]; // only used in onboarding flow
```
Comment (NilsBacke) [inline frontend/src/state/reducers/userReducer.ts:24]: This pattern better reflects how we're using this reducer in conjunction with our api

**Review by imanmoreira — APPROVED** [review]

**Comment by ArunJ98** [comment]
Merging this as it affects the PR I'm working on

**Pull request merged**